test/test_metric/test_metrics.py

```diff
@@ -684,6 +684,56 @@ def test_calculate_losses():
         assert pytest.approx(expected_score) == score_dict[expected_metric]
         assert pytest.approx(1 - expected_score) == loss_dict[expected_metric]
 
+    # Test no metric
+    with pytest.raises(
+        ValueError, match="Number of metrics to compute must be greater than zero."
+    ):
+        calculate_scores(
+            solution=y_true,
+            prediction=y_pred,
+            task_type=BINARY_CLASSIFICATION,
+            metrics=[],
+        )
+
+    with pytest.raises(
+        ValueError, match="Number of metrics to compute must be greater than zero."
+    ):
+        calculate_scores(
+            solution=y_true,
+            prediction=y_pred,
+            task_type=BINARY_CLASSIFICATION,
+            metrics=[],
+            scoring_functions=[
+                autosklearn.metrics.accuracy,
+                autosklearn.metrics.balanced_accuracy,
+            ],
```

Comment:
[Non blocking, later improvement]
Seeing this test now, I'm not sure I fully agree with `calculate_scores`. I would expect to just give it metrics and it gives me answers, i.e. doesn't really make much sense to have a different parameter for both `metrics` and `scoring_functions`.

In the snippet below, `calculate_scores` will `_validate_metrics` but then just merge the two together. I would almost prefer to just move the validation to the surface at `automl.py` and `calculate_scores`/`calculate_losses` just takes a `Sequence[Scorer]` and gives you the scores, doesn't matter if it's the same one 100 times. This also means we don't have to re-perform the validation check everytime we want scores.

https://github.com/automl/auto-sklearn/blob/b1a0c7285af3b563f65ba14f30c5c34f7147a882/autosklearn/metrics/__init__.py#L451-L455

Reply:
That would be an alternative way of doing designing this function. But I agree that we should not do this refactor right now.